### PR TITLE
ad_group_mu_ucn reworked updating members

### PR DESCRIPTION
**Pull request by balcirakpeter**
- Members are now updated in chunks as it will be in ad_mu. Also its
  loging is improved so it will be more readable.
- When all PRs will be merged, subs which are used can be extracted to
  AD script so we prevent duplicities.

### CODE REVIEW

File: send/ad_group_mu_ucn [hunk 1]
```diff
@@ -80,18 +81,20 @@ ldap_unbind($ldap);
...
 print "Removed: " . $counter_remove . " entries.\n";
-print "Updated: " . $counter_update. " entries.\n";
+print "Updated without errors: " . $counter_update . " entries.\n";
+print "Updated with errors: " . $counter_updated_with_errors . " entries.\n";
 print "Failed: " . $counter_fail. " entries.\n";
 
 $lock->unlock();
 
-if ($counter_fail > 0) { die "Failed to process: " . $counter_fail . " entries.\nSee log at: ~/perun-engine/send/logs/$service_name.log";}
+if ($counter_fail > 0 or $counter_updated_with_errors) { die "Some entries failed to process or some members were not updated.\nSee log at: ~/perun-engine/send/logs/$service_name.log";}
```
Comment (stavamichal) [inline send/ad_group_mu_ucn:97]: I would change this to two parts:
```
if ($counter_fail > 0) { die "....";}
if ($counter_updated_with_errors > 0) {die "....";}
```
Message will be more accurate and you will have the info in the printed results.

File: send/ad_group_mu_ucn [hunk 2]
```diff
@@ -214,3 +202,97 @@ sub process_update() {
 	}
 
 }
+
+sub add_members_to_entry {
+
+	my $ad_entry = shift;
+	my $to_be_added = shift;
+	my $return_code = 1;
```
Comment (stavamichal) [inline send/ad_group_mu_ucn:210]: This is correct, but I would prefer something like:
```
our $SUCCESS = 1;
our $FAIL = 0;

my $return_code = $SUCCESS;
```
The same in all places where you are using these return states.

**Review by stavamichal — CHANGES_REQUESTED** [review]

File: send/ad_group_mu_ucn (hunk 1 above)
Comment (balcirakpeter) [reply send/ad_group_mu_ucn]: Done

File: send/ad_group_mu_ucn (hunk 2 above)
Comment (balcirakpeter) [reply send/ad_group_mu_ucn]: Done

**Review by stavamichal — APPROVED** [review]

**Pull request merged**